### ISSUE TITLE
Support for playtesting on theironthrone

### ISSUE THREAD

**Issue by EvadLhorg** [throneteki/throneteki#2958]
Now with the Conclave in place and a maester of design chosen, designing new and redesigning old cards is bound to start. To that end, i think it would be very useful to support playtesting the new stuff online.

Things to discuss that come to my mind to support playtesting:

1. Separate instance of theironthrone.net to play on:
To support a fast cycle of testing, designing and retesting, i think it would be feasible to have a separate instance to deploy and test the new cards on. Maybe a separate github repo/fork might be a good idea to implement new cards quickly
2. Visualizing errata´d/new cards:
We would need some way to display the new card text, challenge icons, STR, traits and so on. Maybe we could expand on the idea already implemented in AltCard.js

Feedback and any ideas welcome!

**Comment by ystros** [comment]
## Card data for new or errata'ed cards

There's a couple directions we could go here:

1. Build out an admin interface that allows you to add / update card data. The cards are already stored in Mongo. To make it easier to use, you could just show cards that have been added or changed through the interface. This would make it very quick to iterate on card stats, but may be tiresome to build out and work with.

2. Bundle card data updates with card implementations. This pre-supposes that testing with card implementations is desirable. You could add a new key on the implementation that specifies the card data directly. This would be slower to iterate on, but it might be alright if you're iterating on abilities

```javascript
class SerPounce {
    // ...
}

SerPounce.code = 'ser-pounce-v2';
SerPournce.cardData = {
    "code": "ser-pounce-v2",
    "type": "character",
    "name": "Ser Pounce",
    "quantity": 3,
    "unique": true,
    "faction": "lannister",
    "loyal": true,
    "cost": 3,
    "icons": {
        "military": false,
        "intrigue": true,
        "power": false
    },
    "strength": 1,
    "traits": [
        "Cat"
    ],
    "text": "...",
    "deckLimit": 3,

};
module.exports = SerPounce;
```

## Manual play testing vs full card implementations

I imagine any new cards will have abilities, so the question becomes: are the designers and play testers alright with manually resolving things at appropriate times, or are full card implementations desired?

Manual resolution could become very tiresome, particularly for Interrupt / Reaction abilities, as the game will not know when to pause for the card. The Interrupt / Reaction problem could be solved by making extremely rudimentary ability configuration when setting card data. But the player would still need to manually pay costs, and resolve the effects as well.

Full implementations would allow for much more fluid play testing, but requires more upfront work (i.e. actually creating an implementation).

## Card modules
If playtesting with full implementations is desirable, then we would want this to occur in a separate repository; both to keep the main repository clean, and to prevent spoilers (if that's a concern). Forking the entire repo and working there would also be inconvenient and could lead to more fracturing.

Instead, it would be better if the server could be configured to pull in cards from a separate repo. This repo could be cloned onto the server, and add a configuration option to pull cards from that directory. Any updates (e.g. pull down new cards) would probably require restarting the game nodes.

Once the Conclave is ready to release the cards, the final implementations could be PR'ed back to the main site repo.

## Where should play testing live?

A separate instance certainly would be the easiest thing to do from our perspective. But it does have the downside of someone having to *run* that separate instance. I don't have a lot of insight on how easy or hard this is these days. It also has the downside of people needing separate accounts (one on play tester instance, one on the main site).

Doing it on the main site would mean needing to segregate the availability of the new cards to only those marked as play testers. If there is an intention to errata old cards, this becomes doubly tricky as the current version needs to remain the same for non-play testers on the site. This *might* be solvable by having a separate game node dedicated to play testing.

Iteration speed and deploy cycle is another factor to consider on the main site. We don't deploy super often currently, and I imagine the designers will want to iterate on at the very minimum a weekly basis. It would be unfortunate if their progress was limited by our schedule.

The final and perhaps biggest concern with allowing play testing on the main site would be security. I don't have a lot of insight into the permissions the code is run with in production, but pulling in full card implementations that we haven't vetted code do a lot of things to compromise the security of the site. The game nodes don't have direct access to the database, but other Bad Things™️ could be done. So we would likely need to vet any cards before they get on the main site
